### PR TITLE
Fix more warnings in the tcs directory

### PR DESCRIPTION
**Pull request by bje-**
An ad-hoc set of fixes for g++ warnings. Sending them in batches!

### CODE REVIEW

File: tcs/cavity_calcs.cpp [hunk 1]
```diff
@@ -141,12 +141,12 @@ void Cavity_Calcs::OuterPanel_Floor( double * F_AF )
 	double Ptheta, Pphi, theta, phi, r1, r2, x, y, y_i, z_i;
 	point p;
 	int hit;
-	int ray_counts = 1;
```
Comment (nickdiorio) [inline tcs/cavity_calcs.cpp]: Do these initializations cause warnings?

File: tcs/csp_solver_core.cpp [hunk 2]
```diff
@@ -3109,9 +3108,6 @@ void C_csp_solver::Ssimulate(C_csp_solver::S_sim_setup & sim_setup)
 				double tol_C = 1.0;								//[C]
 				double tol = tol_C / m_cycle_T_htf_hot_des;		//[-]
 
-				double relaxed_tol_mult = 5.0;				//[-]
```
Comment (nickdiorio) [inline tcs/csp_solver_core.cpp]: These are defined above as "relaxed_tol_multiplier" and "relaxed_tol"

File: tcs/csp_solver_core.cpp [hunk 3]
```diff
@@ -5028,8 +5023,6 @@ void C_csp_solver::Ssimulate(C_csp_solver::S_sim_setup & sim_setup)
 			mc_kernel.mc_sim_info.ms_ts.m_time_start = mc_kernel.mc_sim_info.ms_ts.m_time;	//[s]
 			mc_kernel.mc_sim_info.ms_ts.m_time = mc_kernel.get_baseline_end_time();			//[s]
 			mc_kernel.mc_sim_info.ms_ts.m_step = mc_kernel.mc_sim_info.ms_ts.m_time - mc_kernel.mc_sim_info.ms_ts.m_time_start;	//[s]
-
-			double blah = 1.23;
```
Comment (nickdiorio) [inline tcs/csp_solver_core.cpp]: These were probably used as debugging breakpoints, can always be added if necessary later

File: tcs/heat_exchangers.cpp [hunk 4]
```diff
@@ -781,10 +779,6 @@ void NS_HX_counterflow_eqs::solve_q_dot_for_fixed_UA_enth(int hot_fl_code /*-*/,
 			throw(C_csp_exception("Off-design heat exchanger method failed"));
 		}
 	}
-	else
```
Comment (nickdiorio) [inline tcs/heat_exchangers.cpp]: This control structure isn't needed, it operates in the event that an exception isn't generated, which can be done by simply putting code at the end of the try-catch block

File: tcs/sam_lf_st_pt_type232.cpp [hunk 5]
```diff
@@ -2673,7 +2673,6 @@ void PipeFlowCavity( double Re, double Pr, double LoverD, double relRough, doubl
 	    // given Reynolds number (Re), Prandtl number (Pr), the pipe length diameter ratio (LoverD)
 	    // and the relative roughness
 	    double Gz = Re*Pr/LoverD;	// Eq. 5-79 Nellis and Klein
-	    double x = LoverD/Re;		// Eq. 5-58 Nellis and Klein
```
Comment (nickdiorio) [inline tcs/sam_lf_st_pt_type232.cpp]: I will just comment out, as it's possible it's used for debugging

**Comment by bje-** [comment]
On Mon, Oct 02, 2017 at 05:52:38PM +0000, Nick DiOrio wrote:

> > @@ -141,12 +141,12 @@ void Cavity_Calcs::OuterPanel_Floor( double * F_AF )
>  	double Ptheta, Pphi, theta, phi, r1, r2, x, y, y_i, z_i;
>  	point p;
>  	int hit;
> -	int ray_counts = 1;
> 
> Do these initializations cause warnings? 

No, but I movwd the initialisation of the loop index variable
(ray_counts) into the for loop. It's much clearer and eliminates
warnings about statements with no effect, eg:

  for (ray_counts; ray_counts < max; ray_counts++);

Cheers, Ben

**Comment by nickdiorio** [comment]
Great, thanks!  Just checking for my own understanding.  Your changes have been merged.

**Pull request merged**